### PR TITLE
Stage

### PR DESCRIPTION
**Pull request by Endorf**
- Added support for RTL [ch55904]
- Added possibility to send custom fields in the same request that the SDK CTA  send [ch55802]
- Added tennantId validation [ch57086]

### CODE REVIEW

File: app/src/main/java/com/verygoodsecurity/demoapp/MainActivity.kt [hunk 1]
```diff
@@ -14,7 +15,7 @@ import java.lang.StringBuilder
 
 class MainActivity : AppCompatActivity(), VgsCollectResponseListener, View.OnClickListener {
 
-    val vgsForm = VGSCollect("tntxrsfgxcn")
+    val vgsForm = VGSCollect("tntpszqgikn")
```
Comment (dmytrokhl) [inline app/src/main/java/com/verygoodsecurity/demoapp/MainActivity.kt]: Better not use real tenantId for demo app

File: vgscollect/src/main/java/com/verygoodsecurity/vgscollect/core/api/UrlExtension.kt [hunk 2]
```diff
@@ -11,4 +13,10 @@ internal fun String.setupURL(env:String):String {
         .append(DOMEN)
 
     return builder.toString()
+}
+
+internal fun String.isTennant():Boolean {
+    val m = Pattern.compile("^[tnt][a-zA-Z0-9]*\$").matcher(this)
```
Comment (dmytrokhl) [inline vgscollect/src/main/java/com/verygoodsecurity/vgscollect/core/api/UrlExtension.kt]: Not sure if we need to check that tenantId starts with "tnt"

File: vgscollect/src/main/java/com/verygoodsecurity/vgscollect/view/internal/InputField.kt [hunk 3]
```diff
@@ -242,21 +275,48 @@ internal class EditTextWrapper(context: Context): TextInputEditText(context),
...
-        l?.setBounds(0, 0, cIconSize, cIconSize)
         setCompoundDrawables(l,null,r,null)
     }
 
+    private fun getDrawable(privaryRes:Int?): Drawable? {
+        return privaryRes?.run {
+            val cIconSize = resources.getDimension(R.dimen.c_icon_size).toInt()
+            val d = ContextCompat.getDrawable(context, privaryRes)
+            d?.setBounds(0, 0, cIconSize, cIconSize)
+            d
```
Comment (dmytrokhl) [inline vgscollect/src/main/java/com/verygoodsecurity/vgscollect/view/internal/InputField.kt]: d? - do you need to call it?

File: vgscollect/src/test/java/com/verygoodsecurity/vgscollect/VGSCollectTest.kt [hunk 4]
```diff
@@ -9,28 +9,40 @@ import java.util.regex.Pattern
...
-        val c = VGSCollect("abc", Environment.LIVE)
+        val c = VGSCollect("tntabc", Environment.LIVE)
         assertTrue(c.baseURL.contains("live"))
     }
 
     @Test
     fun testUrl() {
         val regex = "^(https:\\/\\/)+[a-zA-Z0-9 ,]+[.]+[live, sandbox]+[.](verygoodproxy.com)\$"
-        val c = VGSCollect("abc")
+        val c = VGSCollect("tntabc")
```
Comment (dmytrokhl) [inline vgscollect/src/test/java/com/verygoodsecurity/vgscollect/VGSCollectTest.kt]: Do you have tests for FALSE cases? Like "tnt_123", "tenant/tenant"

File: vgscollect/src/main/java/com/verygoodsecurity/vgscollect/core/api/UrlExtension.kt (hunk 2 above)
Comment (flor-master) [inline vgscollect/src/main/java/com/verygoodsecurity/vgscollect/core/api/UrlExtension.kt]: Remove `tnt` prefix from regExp - we have old customers who has custom tnt_id.

File: app/src/main/java/com/verygoodsecurity/demoapp/MainActivity.kt [hunk 5]
```diff
@@ -40,10 +41,19 @@ class MainActivity : AppCompatActivity(), VgsCollectResponseListener, View.OnCli
...
         }
     }
 
+    private fun submitData() {
+        val customData = HashMap<String, String>()
+        customData["nonSDKValue"] = "some additional data"
+
+        val customHeaders = HashMap<String, String>()
+        customHeaders["customHeader"] = "value"
+        vgsForm.asyncSubmit(this@MainActivity, "/post", HTTPMethod.POST, customData, customHeaders)
```
Comment (flor-master) [inline app/src/main/java/com/verygoodsecurity/demoapp/MainActivity.kt]: Please, make setup `customData` and `customHeaders` via setters 
Right now we have breaking changes in our API

File: vgscollect/src/main/java/com/verygoodsecurity/vgscollect/core/api/UrlExtension.kt (hunk 2 above)
Comment (flor-master) [reply vgscollect/src/main/java/com/verygoodsecurity/vgscollect/core/api/UrlExtension.kt]: Remove tnt prefix from regExp - we have old customers who has custom tnt_id.###

File: vgscollect/src/main/java/com/verygoodsecurity/vgscollect/view/InputFieldView.kt [hunk 6]
```diff
@@ -79,16 +79,15 @@ abstract class InputFieldView @JvmOverloads constructor(
     }
 
     override fun setPadding(left: Int, top: Int, right: Int, bottom: Int) {
-        inputField.setPadding(left, top, right, bottom)
-//        super.setPadding(left, top, right, bottom)
+        inputField?.setPadding(left, top, right, bottom)
```
Comment (flor-master) [inline vgscollect/src/main/java/com/verygoodsecurity/vgscollect/view/InputFieldView.kt]: Is it ok - that we setPadding before super method?

Comment (Endorf) [reply vgscollect/src/main/java/com/verygoodsecurity/vgscollect/view/InputFieldView.kt]: it does not affect the parent View.

File: vgscollect/src/test/java/com/verygoodsecurity/vgscollect/VGSCollectTest.kt (hunk 4 above)
Comment (Endorf) [reply vgscollect/src/test/java/com/verygoodsecurity/vgscollect/VGSCollectTest.kt]: yes. In the class VGSCollectTest, method - testInvalidTennantIdURL

File: vgscollect/src/main/java/com/verygoodsecurity/vgscollect/view/internal/InputField.kt (hunk 3 above)
Comment (Endorf) [reply vgscollect/src/main/java/com/verygoodsecurity/vgscollect/view/internal/InputField.kt]: What you mean under 'call'?
In this method i just return Drawable object and scale it if primaryRes is present.

File: app/src/main/java/com/verygoodsecurity/demoapp/JavaActivity.java [hunk 7]
```diff
@@ -53,10 +52,25 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
...
+        vgsForm.resetCustomData();
+        vgsForm.resetCustomHeaders();
+
+        HashMap data = new HashMap<String, String>();
+        data.put("nonSDKValue", "some additional data");
+        vgsForm.setCustomData(data);
+
+        HashMap headers = new HashMap<String, String>();
+        headers.put("CUSTOMHEADER", "value");
+        vgsForm.setCustomHeaders(headers);
```
Comment (dmytrokhl) [inline app/src/main/java/com/verygoodsecurity/demoapp/JavaActivity.java]: Improve naming, if Set is actually Append headers

**Pull request merged**